test/e2e/operator_test.go

```diff
@@ -101,6 +122,80 @@ func TestDefaultClusterIngressExists(t *testing.T) {
 	}
 }
 
+func TestMultipleClusterIngressCreate(t *testing.T) {
+	cl, ns, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	dnsConfig := &configv1.DNS{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, dnsConfig); err != nil {
+		t.Errorf("failed to get DNS 'cluster': %v", err)
+	}
+
+	ing0 := newClusterIngress(clusterIngressName+"0", ns, clusterIngressName+"0"+"."+dnsConfig.Spec.BaseDomain)
+	ing1 := newClusterIngress(clusterIngressName+"1", ns, clusterIngressName+"1"+"."+dnsConfig.Spec.BaseDomain)
+	ingresses := []*ingressv1alpha1.ClusterIngress{ing0, ing1}
+	for _, ing := range ingresses {
+		err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+			if err := cl.Create(context.TODO(), ing); err != nil {
+				if errors.IsAlreadyExists(err) {
+					return true, nil
+				} else {
+					return false, nil
```

Comment:
If I may be picky, I'd suggest simplifying as follows:
```go
			if err := cl.Create(context.TODO(), ing); err != nil {
				if errors.IsAlreadyExists(err) {
					return true, nil
				}
				return false, nil
			}
```
And similar for the other `wait.PollImmediate` callbacks.